iree/compiler/Codegen/LLVMGPU/Passes.cpp

```diff
@@ -25,8 +25,30 @@ static Value gpuAllocationFunction(OpBuilder &builder, Location loc,
                                    ArrayRef<int64_t> staticShape,
                                    Type elementType,
                                    ArrayRef<Value> dynamicSizes) {
+  assert(llvm::all_of(
+             staticShape,
+             [](int64_t dim) { return dim != ShapedType::kDynamicSize; }) &&
+         "Only support static workgroup size right now");
+  // In CUDA workgroup memory is represented by a global variable.
   MemRefType allocType = MemRefType::get(staticShape, elementType, {}, 3);
-  return builder.create<memref::AllocOp>(loc, allocType, dynamicSizes);
+  auto funcOp =
```

Comment:
I would expect that this could be handled during the NVVM lowering by promoting it to a global variable. In effect thats what the CUDA compiler does.

Reply:
> I would expect that this could be handled during the NVVM lowering by promoting it to a global variable. In effect thats what the CUDA compiler does.

Do you mean we should add a pattern to convert AllocOp to globals before conversion to NVVM/LLVM? If we leave the AllocOp it gets converted to malloc by the StandardToLLVM transformations.

Reply:
I added a pattern to convert the AllocOp instead.